pyomo/core/kernel/set_types.py

```diff
@@ -20,232 +20,62 @@
-        else:
-            kwds['name'] = name
-        IntegerSet.__init__(self, **kwds)
-
-Reals=RealSet(name="Reals", doc="A set of real values")
-def validate_PositiveValues(x):    return x >  0
-def validate_NonPositiveValues(x): return x <= 0
-def validate_NegativeValues(x):    return x <  0
-def validate_NonNegativeValues(x): return x >= 0
-def validate_PercentFraction(x):   return x >= 0 and x <= 1.0
-
-PositiveReals    = RealSet(
-  name="PositiveReals",
-  doc="A set of positive real values",
-  validate=validate_PositiveValues,
-  bounds=(0, None)
-)
-NonPositiveReals = RealSet(
-  name="NonPositiveReals",
-  doc="A set of non-positive real values",
-  validate=validate_NonPositiveValues,
-  bounds=(None, 0)
-)
-NegativeReals    = RealSet(
-  name="NegativeReals",
-  doc="A set of negative real values",
-  validate=validate_NegativeValues,
-  bounds=(None, 0)
-)
-NonNegativeReals = RealSet(
-  name="NonNegativeReals",
-  doc="A set of non-negative real values",
-  validate=validate_NonNegativeValues,
-  bounds=(0, None)
-)
-PercentFraction = RealSet(
-  name="PercentFraction",
-  doc="A set of real values in the interval [0,1]",
-  validate=validate_PercentFraction,
-  bounds=(0.0,1.0)
-)
-UnitInterval = RealSet(
-  name="UnitInterval",
-  doc="A set of real values in the interval [0,1]",
-  validate=validate_PercentFraction,
-  bounds=(0.0,1.0)
-)
-
-Integers            = IntegerSet(
-  name="Integers",
-  doc="A set of integer values"
-)
-PositiveIntegers    = IntegerSet(
-  name="PositiveIntegers",
-  doc="A set of positive integer values",
-  validate=validate_PositiveValues,
-  bounds=(1, None)
-)
-NonPositiveIntegers = IntegerSet(
-  name="NonPositiveIntegers",
-  doc="A set of non-positive integer values",
-  validate=validate_NonPositiveValues,
-  bounds=(None, 0)
-)
-NegativeIntegers    = IntegerSet(
-  name="NegativeIntegers",
-  doc="A set of negative integer values",
-  validate=validate_NegativeValues,
-  bounds=(None, -1)
-)
-NonNegativeIntegers = IntegerSet(
-  name="NonNegativeIntegers",
-  doc="A set of non-negative integer values",
-  validate=validate_NonNegativeValues,
-  bounds=(0, None)
-)
-
-Boolean = BooleanSet(name="Boolean", doc="A set of boolean values")
-Binary  = BooleanSet(name="Binary",  doc="A set of boolean values")
+#
+# Dummy types used by Kernel as domain flags
+#
+class RealSet(object):
+    @staticmethod
+    def get_interval():
+        return (None, None, 0)
+
+    @staticmethod
+    def is_continuous():
+        return True
+
+    @staticmethod
+    def is_integer():
+        return False
+
+    @staticmethod
+    def is_binary():
+        return False
+
+
+class IntegerSet(object):
+    @staticmethod
+    def get_interval():
+        return (None, None, 1)
+
+    @staticmethod
+    def is_continuous():
+        return False
+
+    @staticmethod
+    def is_integer():
+        return True
+
+    @staticmethod
+    def is_binary():
+        return False
+
+
+class BinarySet(object):
```

Comment:
The BinarySet, BooleanSet objects don't seem to serve a purpose in kernel.

Reply:
I agree, but I think I ran into examples that used BinarySet and BooleanSet from kernel.  Regardless, I left them in for backwards compatibility.  If we can later confirm that they are not necessary (and no one is using them), then we can delete.

Reply:
I'm fairly certain this these are not needed in kernel. Binary, yes. BinarySet, no.

Reply:
BooleanSet is used in `pyomo/core/tests/unit/kernel/test_variable.py` (and historically `BooleanSet` and `BinarySet` have been the same class), but it was not imported into `pyomo.kernel`.

Part of my refactoring was so that `kernel` proper only needs `RealSet` and `IntegerSet`, and can infer Binary from a standard `get_interval()` method.  Longer term, I would like to standardize on `BinarySet` where needed so that we can re-purpose `BooleanSet` to be a proper Boolean (i.e., True/False, and not 0/1) set.